compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -366,14 +366,28 @@ fn unreachable_pattern(tcx: TyCtxt<'_>, span: Span, id: HirId, catchall: Option<
 }
 
 fn irrefutable_let_pattern(tcx: TyCtxt<'_>, span: Span, id: HirId, source: hir::MatchSource) {
-    tcx.struct_span_lint_hir(IRREFUTABLE_LET_PATTERNS, id, span, |lint| {
-        let msg = match source {
-            hir::MatchSource::IfLetDesugar { .. } => "irrefutable if-let pattern",
-            hir::MatchSource::WhileLetDesugar => "irrefutable while-let pattern",
-            hir::MatchSource::IfLetGuardDesugar => "irrefutable if-let guard",
-            _ => bug!(),
-        };
-        lint.build(msg).emit()
+    tcx.struct_span_lint_hir(IRREFUTABLE_LET_PATTERNS, id, span, |lint| match source {
+        hir::MatchSource::IfLetDesugar { .. } => {
+            let mut diag = lint.build("irrefutable if-let pattern");
+            diag.note("this pattern will always match, so the if-let is useless");
+            diag.help("consider replacing the if-let with a let");
```

Comment:
```suggestion
            diag.help("consider replacing the if-let with a `let`");
```
I wonder whether it would be more readable to write
```suggestion
            diag.help("consider replacing the `if let` with a `let`");
```
as well.

Reply:
I intentionally removed the backticks on `let` to be consistent with if-let. I thought of doing the second option you suggest, but didn't because then it would be inconsistent with the error message. I'm still open to it, I just want to make sure we don't confuse people. What do you think?

Reply:
I'd be tempted to replace `if-let` with ``` `if let` ``` in general (including in the main error message). To me, it's slightly easier to read, and feels more consistent with other uses of code fragments in error messages.

Reply:
I like the look of `if-let`, though I do think it's probably clearer as `` `if let` ``. However, it seems like `if-let` and the like are used *a lot*, not just in the compiler but also in documentation. So can we do that as a separate change so this one doesn't wallow too long?

Reply:
That works (as long as we don't overlook that `let` needs to be enclosed in backticks in the message above as well).